src/Gaffer/Reference.cpp

```diff
@@ -69,10 +69,72 @@ using namespace Gaffer;
 namespace
 {
 
+bool descendantHasInput( const Plug *plug )
+{
+	for( auto &d : Plug::RecursiveRange( *plug ) )
+	{
+		if( d->getInput() )
+		{
+			return true;
+		}
+	}
+	return false;
+}
+
+bool conformSplinePlugs( const Gaffer::Plug *srcPlug, Gaffer::Plug *dstPlug, bool ignoreDefaultValues )
+{
+	auto conform = [=] ( auto typedSrc, Gaffer::Plug *dst ) {
+
+		using PlugType = std::remove_const_t<std::remove_pointer_t<decltype( typedSrc )>>;
+		auto typedDest = runTimeCast<PlugType>( dst );
+		if( !typedDest )
+		{
+			return false;
+		}
+
+		if( typedSrc->isSetToDefault() && ignoreDefaultValues && !descendantHasInput( typedSrc ) )
+		{
+			// No need to conform, since we don't want to transfer anything.
```

Comment:
Am I correct that this is what ensures that we can pick up updates from upstream changes to the reference if we haven't changed anything?  If so, "No need" seems to be an understatement, which makes it sound like just an optimization.  We need to not conform if there are no changes.

Reply:
Yep, that's correct. Would "We won't be transferring anything, so we shouldn't conform" be better?

Reply:
Yeah, that sounds better.  Or even something like, "Since it hasn't been overridden, we will keep the original plugs and plug values"

Reply:
I added something that is hopefully better in https://github.com/GafferHQ/gaffer/pull/5205/commits/0f2b5dbaf922c69134ed2b332ac59689ec092f64.

Reply:
Sure, that sounds good to me.